addons/mail/models/mail_activity.py

```diff
@@ -373,7 +373,8 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
                 doc_operation = 'read'
             DocumentModel = self.env[doc_model].with_user(access_rights_uid or self._uid)
             right = DocumentModel.check_access_rights(doc_operation, raise_exception=False)
-            if right:
+            doc_ids.discard(0)  # in case it was used as a hack to signal no linked record
+            if right and doc_ids:
```

Comment:
As anyway activity_to_documents is used only here, shoudln't we remove 0 directly above when computing activity_to_documents ? Anyway if we have res_id = 0, they will not be in allowed_ids, and no need to check the access rights on the model itself.

Reply:
I think then we could patch the query
https://github.com/odoo/odoo/blob/eb8025a2cf605fd8cc415656f6c0663abce25092/addons/mail/models/mail_activity.py#L357-L359
as
```sql
                SELECT DISTINCT activity.id, activity.res_model, activity.res_id
                FROM "%s" activity
                WHERE activity.id = ANY (%%(ids)s) AND activity.res_id!=0""" % self._table, dict(ids=list(sub_ids)))
```

Does it look correct?

Reply:
And to be complete, we can also ignore NULL values
```sql
AND COALESCE(activity.res_id, 0) != 0
```

Reply:
You're right.